axis.go

```diff
@@ -497,6 +498,158 @@ func (utt UnixTimeTicks) Ticks(min, max float64) []Tick {
+	{30 * time.Second,
+		"15:04:05", "15:04", ":05", 30 * time.Second},
+	{time.Minute,
+		"15:04:05", "15:04", ":05", time.Minute},
+	{2 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 2 * time.Minute},
+	{5 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 5 * time.Minute},
+	{15 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 15 * time.Minute},
+	{30 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 30 * time.Minute},
+	{time.Hour,
+		"Jan 2, 3pm", "Jan 2", "3pm", time.Hour},
+	{3 * time.Hour,
+		"Jan 2, 3pm", "Jan 2", "3pm", 3 * time.Hour},
+	{6 * time.Hour,
+		"Jan 2, 3pm", "Jan 2", "3pm", 6 * time.Hour},
+	{12 * time.Hour,
+		"Jan 2", "Jan 2", "3pm", 12 * time.Hour},
+	{24 * time.Hour,
+		"Jan 2", "Jan", "2", 24 * time.Hour},
+	{48 * time.Hour,
+		"Jan 2", "Jan", "2", 48 * time.Hour},
+	{7 * 24 * time.Hour,
+		"Jan 2", "Jan", "2", 7 * 24 * time.Hour},
+	{aMonth, // 1 month
+		"Jan 2006", "2006", "Jan", aMonth},
+	{3 * aMonth, // 3 months
+		"Jan 2006", "2006", "Jan", 3 * aMonth},
+	{6 * aMonth, // 6 months
+		"Jan 2006", "2006", "Jan", 6 * aMonth},
+	{12 * aMonth, // 1 year
+		"2006", "", "2006", 12 * aMonth},
+	{2 * 12 * aMonth, // 2 years
+		"2006", "", "2006", 2 * 12 * aMonth},
+	{5 * 12 * aMonth, // 5 years
+		"2006", "", "2006", 5 * 12 * aMonth},
+	{10 * 12 * aMonth, // 10 years
+		"2006", "", "2006", 10 * 12 * aMonth},
+}
+
+var aMonth = 31 * 24 * time.Hour
+
+type tickRule struct {
+	DurationPerInch time.Duration // use this rule for a maximum Duration per inch
+	LongFormat      string        // longer format
+	WatchFormat     string        // show long format when WatchFormat changes between ticks
+	ShortFormat     string        // incremental format, shorter
+	TimeWindow      time.Duration // interval for ticks
+}
+
+// Ticks implements plot.Ticker.
+func (t AutoUnixTimeTicks) Ticks(min, max float64) []Tick {
+	width := t.Width
+	if width == 0 {
+		width = 4 * vg.Inch
+	}
+
+	minT := time.Unix(int64(min), 0).UTC()
+	maxT := time.Unix(int64(max), 0).UTC()
+	durationPerInch := maxT.Sub(minT) / time.Duration(width/vg.Inch)
+
+	rule := tickIntervals[len(tickIntervals)-1]
+	for idx := range tickIntervals[:len(tickIntervals)-1] {
+		if durationPerInch < tickIntervals[idx+1].DurationPerInch {
+			rule = tickIntervals[idx]
+			break
+		}
+	}
+
+	ticks := []Tick{
+	//	{Value: min, Label: minT.Format(rule.BeginFormat)},
+	}
+
+	monthsDelta := time.Month(rule.TimeWindow / aMonth)
+
+	//fmt.Println("Months delta", int(monthsDelta))
+	start := minT.Truncate(rule.TimeWindow)
+	count := 0
+	lastWatch := ""
+	for {
+		count++
+		if monthsDelta > 0 {
+			// Count in Months now
+			start = time.Date(start.Year(), start.Month()+monthsDelta, 1, 0, 0, 0, 0, time.UTC)
+		} else {
+			start = start.Add(rule.TimeWindow)
+		}
+
+		if count > 20 {
+			os.Exit(0)
+		}
+
+		if start.Before(minT) {
+			continue
+		}
+		if start.After(maxT) {
+			break
+		}
+
+		label := ""
+		newWatch := start.Format(rule.WatchFormat)
```

Comment:
`watch` is probably not the right word... it is the format we check for changes, to decide if we display the long form (like when we change from `Sep 1` to `Sep 2` when showing hours), or the short (like `11pm`, then `Sep 2, 12pm` when `watch` changes)

If you look into tests, you'll see there are 2 sorts of labels per graph.. long and short forms.